enhancements/monitoring/scrape-profiles.md

```diff
@@ -0,0 +1,407 @@
+spec:
+  endpoints:
+  - bearerTokenFile: /var/run/secrets/kubernetes.io/serviceaccount/token
+    interval: 30s
+    port: https
+    scheme: https
+    tlsConfig:
+      <...>
+  jobLabel: k8s-app
+  selector:
+    matchLabels:
+      k8s-app: telemeter-client
+  metricRelabelings:
+  - sourceLabels: [__name__]
+    action: keep
+    regex: "federate_samples|federate_filtered_samples"
+ ```
+
+Note: the `metricRelabeling` section only keeping two metrics, while the rest is
+dropped.
+
+Finally, a team that adopts the scrape profile feature should also add 
+themselves and their component to the list under 
+[Infrastruture needed](#infrastructure-needed-optional)
+
+#### Variation [optional]
+
+NA
+
+### API Extensions
+
+- Addition of a new field to the `cluster-monitoring-config` ConfigMap
+
+```go
+type PrometheusK8sConfig struct {
+  // Defines the scraping profile that will be enforced on the platform
+  // prometheus instance. Possible values are full and minimal.
+	ScrapeProfile string `json:"scrapeProfile,omitempty"`
+}
+```
+
+### Implementation Details/Notes/Constraints
+
+Each OpenShift team that wants to adopt this feature will be responsible for
+providing the different monitors. This work is not trivial. Dependencies between
+operators and their metrics exist. This makes it difficult for developers to
+determine whether a given metric must be added to the "minimal" profile or not.
+To aid teams with this problem we will provide a tool that would consume a
+monitor and generate the "minimal" monitor for that operator. However for this,
+the tool would need to have access to an up to date, installation of OpenShift
+in order to query the Prometheus instance running in the cluster.
+
+### Risks and Mitigations
+
+- How are monitors supposed to be kept up to date? In 4.12 a metric that wasn't
+  being used in an alert is now required, how does the monitor responsible for
+  that metric gets updated?
+  - The tool we provide would run in CI and ensure are ServiceMonitors are up to
+    date;
+
+- A new profile is added, what will happen to operators that had implemented
+  scrape profiles but did not implement the latest profile.
+  - The monitoring team will use the list under 
+  [Infrastruture needed](#infrastructure-needed-optional) to help the developers
+  with the addoption of the new scrape profile.
+
+### Drawbacks
+
+- Extra CI cycles
+
+## Design Details
+
+### Open Questions
+
+- Should we add future profiles? 
+- How will ensure that all metrics used by the console are still present?
+- What happens if CMO pick-up an unsupported scrape value? (dicussion on
+  https://github.com/openshift/enhancements/pull/1298#discussion_r1072266853)
+
+### Test Plan
+
+- Unit tests in CMO to validate that the correct monitors are being selected
+- E2E tests in CMO to validate that everything works correctly Unsure on this
+one... (- Testing in openshift/CI to validate that every metrics being used
+exists in the cluster?)
```

Comment:
```suggestion
- E2E tests in CMO to validate that everything works correctly.
```
> [...] every metrics being used exists in the cluster?

Expanding on the comment above, there can two cases to this. For internal metrics, we can maintain a list of all required metrics, whereas for external metrics, these can be pushed to CMO by the respective team (single source of truth). The operator will validate them all the same.

Reply:
Same as https://github.com/openshift/enhancements/pull/1298#discussion_r1267919544